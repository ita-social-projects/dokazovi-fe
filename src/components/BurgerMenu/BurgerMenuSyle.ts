import makeStyles from '@material-ui/core/styles/makeStyles';

export const useStyles = makeStyles((theme) => ({
  icon: {
    color: theme.palette.common.white,
  },
  menuIcon: {
    fontSize: 48,
  },
  header: {
    fontSize: 32,
    marginTop: theme.spacing(8),
  },
  list: {
    margin: 0,
    padding: 0,
    backgroundColor: 'black',
    color: 'white',
    height: '100%',
    border: 'none',
    width: `207px`,
    paddingLeft: theme.spacing(5),
  },
  link: {
    display:'block',
    marginTop: theme.spacing(8),
    color: theme.palette.primary.main,
    fontWeight: 600,
    '&.active': {
      fontWeight: 800,
      color: theme.palette.common.white,
    },
  },
  headerLinks: {
    fontSize: 22,
  },
  footerLinks: {
    fontWeight: 500,
    fontSize: 16,
    lineHeight: '18px',
  },
  root:{
    backgroundColor:theme.palette.common.black,
  },
}));
