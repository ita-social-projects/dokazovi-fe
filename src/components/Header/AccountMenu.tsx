import React, { useContext, useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { Avatar, Button, Typography } from '@material-ui/core';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import { useTranslation } from 'react-i18next';
import { useStyles } from './AccountMenu.styles';
import {
  StyledMenu,
  StyledMenuItem,
} from '../../old/lib/components/Menu/StyledMenu';
import {
  getUserAsyncAction,
  selectCurrentUser,
  signOutAction,
} from '../../models/user';
import { useActions } from '../../shared/hooks';
import { AuthContext } from '../../old/provider/AuthProvider/AuthContext';
import { AccountIcon } from '../../old/lib/components/icons/AccountIcon';
import { langTokens } from '../../locales/localizationInit';
import { clearAuthoritiesAction } from '../../models/authorities/reducers';
import { useCheckAdmin } from '../../old/lib/hooks/useCheckAdmin';

export const AccountMenu: React.FC = () => {
  const { t } = useTranslation();

  const classes = useStyles();
  const user = useSelector(selectCurrentUser);
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
  const { authenticated } = useContext(AuthContext);
  const { removeAuthorization } = useContext(AuthContext);
  const [boundSignOutAction] = useActions([signOutAction]);
  const [boundGetUserAsyncAction] = useActions([getUserAsyncAction]);
  const [boundClearAuthorities] = useActions([clearAuthoritiesAction]);
  const authorities = useCheckAdmin();

  const onLogoutHandler = () => {
    boundSignOutAction();
    removeAuthorization();
    boundClearAuthorities();
  };

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  useEffect(() => {
    if (authenticated) {
      boundGetUserAsyncAction();
    }
  }, [authenticated]);

  if (!user.data) {
    return null;
  }

  return (
    <>
      <Button
        color="primary"
        aria-controls="simple-menu"
        aria-haspopup="true"
        onClick={handleClick}
        className={classes.button}
      >
        {user.data ? (
          <Avatar
            alt={`${user.data.firstName} ${user.data.lastName}`}
            className={classes.avatar}
            src={user.data.avatar}
          />
        ) : (
          <AccountIcon className={classes.avatar} />
        )}
        {user.data && <ArrowDropDownIcon className={classes.icon} />}
      </Button>
      <StyledMenu
        id="account-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
        className={classes.menu}
      >
        <Link to="/profile">
          <StyledMenuItem onClick={handleClose}>
            <Typography variant="button" color="inherit">
              {t(langTokens.common.profile)}
            </Typography>
          </StyledMenuItem>
        </Link>
        {authorities ? (
          <Link to="/admin">
            <StyledMenuItem onClick={handleClose}>
              <Typography variant="button" color="inherit">
                {t(langTokens.common.admin)}
              </Typography>
            </StyledMenuItem>
          </Link>
        ) : null}
        <Link to="/">
          <StyledMenuItem onClick={onLogoutHandler}>
            <Typography variant="button" color="inherit">
              {t(langTokens.common.exit)}
            </Typography>
          </StyledMenuItem>
        </Link>
      </StyledMenu>
    </>
  );
};
