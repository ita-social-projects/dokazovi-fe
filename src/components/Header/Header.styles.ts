import makeStyles from '@material-ui/core/styles/makeStyles';

export const useStyles = makeStyles(
  (theme) => ({
    header: {
      height: 120,
      backgroundColor: theme.palette.common.black,
    },
    headerMobile: {
      height: 85,
      backgroundColor: theme.palette.common.black,
    },
    container: {
      height: '100%',
      padding: theme.spacing(0, 10),
      [theme.breakpoints.down('xs')]: {
        padding: theme.spacing(0, 5),
      },
    },
    toolbar: {
      height: '100%',
      justifyContent: 'space-between',
      padding: 0,
    },
    logo: {
      color: theme.palette.common.white,
      marginBottom: 6,
    },
    actionsContainer: {
      display: 'flex',
      columnGap: theme.spacing(4),
    },
    tabs: {
      height: 55,
      marginLeft: '6vw',
      display: 'flex',
      alignItems: 'center',
      backgroundColor: theme.palette.common.black,
      [theme.breakpoints.down('md')]: {
        margin: '0 3vw 0 4vw',
      },
    },
    tab: {
      color: theme.palette.primary.main,
      fontWeight: 600,
      '&:not(:last-child)': {
        marginRight: theme.spacing(5),
      },
      '&.active': {
        color: theme.palette.common.white,
        fontWeight: 700,
      },
    },
    tabLabel: {
      fontSize: 18,
      color: 'inherit',
      fontWeight: 'inherit',
    },
    logoMobile: {
      fontSize: 38,
      color: theme.palette.common.white,
    },
    searchIcon: {
      color: theme.palette.common.white,
      fontSize: 48,
    },
    searchInputWrapper: {
      padding: 0,
    },
    searchInput: {
      width: '85%',
      height: 39,
      fontSize: 16,
      paddingLeft: theme.spacing(2),
    },
    searchInputIcon: {
      backgroundColor: theme.palette.common.white,
      marginRight: theme.spacing(2),
      fontSize: 32,
    },
    paper: {
      width: '70%',
    },
    label: {
      padding: 0,
    },
  }),
  { name: 'Header' },
);
