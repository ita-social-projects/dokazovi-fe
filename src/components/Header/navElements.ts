import { IHeaderProps } from './types';
import i18n, { langTokens } from '../../locales/localizationInit';

export const navElements: IHeaderProps[] = [
  {
    id: 'main',
    label: i18n.t(langTokens.common.main),
    url: '/',
  },
  {
    id: 'materials',
    label: i18n.t(langTokens.common.materials),
    url: '/materials',
  },
  {
    id: 'experts',
    label: i18n.t(langTokens.common.experts),
    url: '/experts',
  },
];
