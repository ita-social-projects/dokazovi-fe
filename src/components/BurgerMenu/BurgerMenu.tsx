import React from 'react';
import {
  Drawer,
  Hidden,
  IconButton,
  List,
  Typography,
} from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import { useTranslation } from 'react-i18next';
import { NavLink } from 'react-router-dom';
import { langTokens } from '../../locales/localizationInit';
import { IHeaderProps } from '../Header/types';
import { useStyles } from './BurgerMenuSyle';
import { navElements } from '../../old/lib/components/Footer/Footer';

export interface IBurgerMenuProps {
  navigation: IHeaderProps[];
  mobileMenuOpen: boolean;
  setMobileMenuOpen: (b: boolean) => void;
}

export const BurgerMenu: React.FC<IBurgerMenuProps> = ({
  navigation,
  setMobileMenuOpen,
  mobileMenuOpen,
}) => {
  const classes = useStyles();
  const { t } = useTranslation();

  const handleDrawerToggle = () => {
    setMobileMenuOpen(!mobileMenuOpen);
  };

  return (
    <div>
      <IconButton
        data-testid="menuButton"
        className={classes.icon}
        aria-label="Open drawer"
        edge="start"
        onClick={handleDrawerToggle}
      >
        <MenuIcon className={classes.menuIcon} />
      </IconButton>
      <Hidden>
        <Drawer
          classes={{ paper: classes.root }}
          variant="temporary"
          anchor="left"
          open={mobileMenuOpen}
          onClose={handleDrawerToggle}
          ModalProps={{
            keepMounted: true,
          }}
        >
          <div className={classes.list}>
            <Typography className={classes.header} variant="h1">
              {t(langTokens.common.projectName)}
            </Typography>
            <List>
              {navigation.map((item) => (
                <NavLink
                  className={classes.link}
                  onClick={handleDrawerToggle}
                  to={item.url}
                  key={item.id}
                  exact
                >
                  <Typography className={classes.headerLinks} variant="h5">
                    {item.label}
                  </Typography>
                </NavLink>
              ))}
              {navElements.map((item) => (
                <NavLink
                  className={classes.link}
                  onClick={handleDrawerToggle}
                  to={item.url}
                  key={item.id}
                  exact
                >
                  <Typography className={classes.footerLinks} component="span">
                    {item.label}
                  </Typography>
                </NavLink>
              ))}
            </List>
          </div>
        </Drawer>
      </Hidden>
    </div>
  );
};
