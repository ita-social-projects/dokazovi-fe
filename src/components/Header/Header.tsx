import React, { useContext, useState } from 'react';
import {
  Box,
  Button,
  Container,
  Menu,
  MenuItem,
  Toolbar,
  Typography,
} from '@material-ui/core';
import { Link, NavLink, Route } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import SearchIcon from '@material-ui/icons/Search';
import { useStyles } from './Header.styles';
import { PostCreationMenu } from './PostCreationMenu';
import { LoginModal } from '../../old/lib/components/Users/LoginModal';
import { AccountMenu } from './AccountMenu';
import { AuthContext } from '../../old/provider/AuthProvider/AuthContext';
import i18n, { langTokens } from '../../locales/localizationInit';
import { IHeaderProps } from './types';
import { ScreenContext } from '../../old/provider/MobileProvider/ScreenContext';
import { BurgerMenu } from '../BurgerMenu/BurgerMenu';

export const navElems: IHeaderProps[] = [
  {
    id: 'main',
    label: i18n.t(langTokens.common.main),
    url: '/',
  },
  {
    id: 'materials',
    label: i18n.t(langTokens.common.materials),
    url: '/materials',
  },
  {
    id: 'experts',
    label: i18n.t(langTokens.common.experts),
    url: '/experts',
  },
];

export const Header: React.FC = () => {
  const classes = useStyles();
  const { t } = useTranslation();
  const { authenticated } = useContext(AuthContext);

  const [searchInput, setSearchInput] = useState<null | Element>(null);
  const [mobileMenuOpen, setMobileMenuOpen] = useState<boolean>(false);

  const { mobile } = useContext(ScreenContext);

  return (
    <div id="header" className={classes.header}>
      <Container className={classes.container}>
        <Toolbar className={classes.toolbar}>
          {mobile && <BurgerMenu
            navigation={navElems}
            mobileMenuOpen={mobileMenuOpen}
            setMobileMenuOpen={(b)=>setMobileMenuOpen(b)}
          />}
          <Box display="flex">
            {!mobileMenuOpen && <Link to='/'>
              <Typography className={mobile ? classes.logoMobile : classes.logo} variant='h1'>
                {t(langTokens.common.projectName)}
              </Typography>
            </Link>}
            {!mobile && <Box className={classes.tabs}>
              {navElems.map((item) => (
                <NavLink
                  to={item.url}
                  key={item.id}
                  className={classes.tab}
                  exact
                >
                  <Typography variant='h5' className={classes.tabLabel}>
                    {item.label}
                  </Typography>
                </NavLink>
              ))}
            </Box>}
          </Box>
          {!mobile && <Box className={classes.actionsContainer}>
            {authenticated && <PostCreationMenu />}
            {authenticated ? (
              <AccountMenu />
            ) : (
              <Route path='/opendoctorgate'>
                <LoginModal />
              </Route>
            )}
          </Box>}
          {mobile && <div>
            <Button aria-controls="simple-menu" aria-haspopup="true"
                    onClick={(e)=>setSearchInput(e.currentTarget)}>
              <SearchIcon className={classes.searchIcon} />
            </Button>
            <Menu
              id="simple-menu"
              anchorEl={searchInput}
              keepMounted
              open={Boolean(searchInput)}
              onClose={()=>setSearchInput(null)}
            >
              <MenuItem>
                <input className={classes.searchInput}/>
                <SearchIcon/>
              </MenuItem>
            </Menu>
          </div>}
        </Toolbar>
      </Container>
    </div>
  );
};
