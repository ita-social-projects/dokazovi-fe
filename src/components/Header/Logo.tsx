import React from 'react';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { langTokens } from '../../locales/localizationInit';
import { useStyles } from './Header.styles';

export const Logo = ({ mobile, isSearchVisible }) => {
  const { t } = useTranslation();
  const classes = useStyles();
  if (isSearchVisible) return null;

  return (
    <>
      <Box display="flex">
        <Link to="/">
          <Typography
            className={mobile ? classes.logoMobile : classes.logo}
            variant="h1"
          >
            {t(langTokens.common.projectName)}
          </Typography>
        </Link>
      </Box>
    </>
  );
};
