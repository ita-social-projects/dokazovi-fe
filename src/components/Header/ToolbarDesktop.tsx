import React, { useContext } from 'react';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import { NavLink, Route } from 'react-router-dom';
import { useStyles } from './Header.styles';
import { AuthContext } from '../../old/provider/AuthProvider/AuthContext';
import { PostCreationMenu } from './PostCreationMenu';
import { AccountMenu } from './AccountMenu';
import { LoginModal } from '../../old/lib/components/Users/LoginModal';
import { Logo } from './Logo';
// eslint-disable-next-line import/no-cycle
import { navElems } from './Header';

export const ToolbarDesktop = () => {
  const classes = useStyles();
  const { authenticated } = useContext(AuthContext);
  const mobile = false;
  const isSearchVisible = false;

  return (
    <div className={classes.toolbarDesktop}>
      <div style={{ display: 'flex' }}>
        <Logo mobile={mobile} isSearchVisible={isSearchVisible} />
        <Box className={classes.tabs}>
          {navElems.map((item) => (
            <NavLink to={item.url} key={item.id} className={classes.tab} exact>
              <Typography variant="h5" className={classes.tabLabel}>
                {item.label}
              </Typography>
            </NavLink>
          ))}
        </Box>
      </div>
      <Box className={classes.actionsContainer}>
        {authenticated && <PostCreationMenu />}
        {authenticated ? (
          <AccountMenu />
        ) : (
          <Route path="/opendoctorgate">
            <LoginModal />
          </Route>
        )}
      </Box>
    </div>
  );
};
