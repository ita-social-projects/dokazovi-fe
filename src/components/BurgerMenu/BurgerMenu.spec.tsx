import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BurgerMenu } from './BurgerMenu';
import { navElements } from '../Header/navElements';

describe('Burger menu test', () => {
  it('should render properly', () => {
    const { asFragment } = render(
      <BurgerMenu
        navigation={navElements}
        mobileMenuOpen
        setMobileMenuOpen={jest.fn}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should invoke function on click', () => {
    const clickHandlerMock = jest.fn();
    render(
      <BurgerMenu
        navigation={navElements}
        mobileMenuOpen
        setMobileMenuOpen={clickHandlerMock}
      />,
    );
    const menuButton = screen.getByTestId('menuButton');
    userEvent.click(menuButton);
    expect(clickHandlerMock).toHaveBeenCalled();
  });
});
