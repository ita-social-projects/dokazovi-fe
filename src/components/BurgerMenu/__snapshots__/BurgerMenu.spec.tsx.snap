// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Burger menu test should render properly 1`] = `
<DocumentFragment>
  <div>
    <button
      aria-label="Open drawer"
      class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-1 MuiIconButton-edgeStart"
      data-testid="menuButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-menuIcon-2"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
